pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -82,468 +105,215 @@ func NewKubeConfig(opts *Options) (*KubeConfig, error) {
 		}
-	} else {
-		// Let's request the KubeConfig from the virtual workspaces
-		workspaceKubeConfigBytes, err := tenancyClient.TenancyV1beta1().RESTClient().Get().Resource("workspaces").SubResource("kubeconfig").Name(requestedWorkspaceName).Do(ctx).Raw()
+		u.Path = path.Join(u.Path, parentClusterName.Path())
+		newServerHost = u.String()
+
+	case "":
+		return kc.CurrentWorkspace(ctx)
+
+	default:
+		config, err := clientcmd.NewDefaultClientConfig(*kc.startingConfig, kc.overrides).ClientConfig()
 		if err != nil {
 			return err
 		}
-		workspaceConfig, err := clientcmd.Load(workspaceKubeConfigBytes)
+		u, currentClusterName, err := parseClusterURL(config.Host)
 		if err != nil {
-			return err
+			return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
 		}
-		newContextCluster = workspaceConfig.Clusters[workspaceConfig.CurrentContext]
-	}
 
-	// Update the current workspace cluster info with the new cluster info
-	kc.startingConfig.Clusters[kcpCurrentWorkspaceContextKey] = newContextCluster.DeepCopy()
+		if strings.Contains(name, ":") || name == tenancyv1alpha1.RootCluster.String() {
+			// absolute logical cluster
+			u.Path = path.Join(u.Path, logicalcluster.New(name).Path())
+			newServerHost = u.String()
+		} else {
+			// relative logical cluster, get URL from workspace object in current context
+			ws, err := kc.client.Cluster(currentClusterName).TenancyV1beta1().Workspaces().Get(ctx, name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if ws.Status.Phase != tenancyv1alpha1.ClusterWorkspacePhaseReady {
+				return fmt.Errorf("workspace %q is not ready", name)
+			}
 
-	// Add authentication info based on either the kubectl overrides or the current or previous kubectl context
-	newContext, err := kc.getContext(contextName)
-	if err != nil {
-		return err
-	}
-	authInfoName := newContext.AuthInfo
-	if authInfoName == "" {
-		authInfoName = currentContext.AuthInfo
-	}
-	kubectlOverrides := opts.KubectlOverrides
-	if !reflect.ValueOf(kubectlOverrides.AuthInfo).IsZero() {
-		kc.startingConfig.AuthInfos[kcpOverridesContextAuthInfoKey] = &kubectlOverrides.AuthInfo
-		authInfoName = kcpOverridesContextAuthInfoKey
-	} else {
-		if authInfoName != kcpOverridesContextAuthInfoKey && currentContextAuthName != kcpOverridesContextAuthInfoKey {
-			delete(kc.startingConfig.AuthInfos, kcpOverridesContextAuthInfoKey)
+			newServerHost = ws.Status.URL
+			workspaceType = ws.Spec.Type
 		}
 	}
 
-	kc.startingConfig.Contexts[kcpCurrentWorkspaceContextKey] = &api.Context{
-		Cluster:  kcpCurrentWorkspaceContextKey,
-		AuthInfo: authInfoName,
+	// modify kubeconfig, using the "workspace" context and cluster
+	newKubeConfig := kc.startingConfig.DeepCopy()
+	oldCluster, found := kc.startingConfig.Clusters[currentContext.Cluster]
+	if !found {
+		return fmt.Errorf("cluster %q not found in kubeconfig", currentContext.Cluster)
 	}
+	newCluster := *oldCluster
```

Comment:
So you don't manage AuthInfo at all in the current and previous context ?
Afair in previous tests I had done, it might bring some very painful behavior where you cannot bring back the auth used (through a --token override for example) with the previous workspace.

Reply:
Yes I think this needs to be carried forward

Reply:
Based on @sttts 's comment in Slack
>verified that overrides don't affect  what is persisted. This is what kubectl itself does too. overrides are only used to tweak a client used by kubectl/the plugin.

👍